src/SecTester.Scan/ScanSettings.cs

```diff
@@ -0,0 +1,199 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using SecTester.Scan.Models;
+using SecTester.Scan.Target;
+
+namespace SecTester.Scan;
+
+public record ScanSettings : ScanSettingsOptions
+{
+  private const int MinNameLength = 1;
+  private const int MaxNameLength = 200;
+  private const int MinPoolSize = 1;
+  private const int MaxPoolSize = 50;
+  private const bool DefaultSmart = true;
+  private const bool DefaultSkipStaticParams = true;
+  private const int DefaultPoolSize = 10;
+
+  private static readonly TimeSpan MinTargetTimeout = TimeSpan.FromSeconds(1);
+  private static readonly TimeSpan MaxTargetTimeout = TimeSpan.FromSeconds(120);
+  private static readonly TimeSpan MinSlowEpTimeout = TimeSpan.FromSeconds(100);
+  private static readonly TimeSpan DefaultTargetTimeout = TimeSpan.FromSeconds(5);
+  private static readonly TimeSpan DefaultSlowEpTimeout = TimeSpan.FromSeconds(1000);
+  private static readonly IEnumerable<AttackParamLocation> DefaultAttackParamLocations = new List<AttackParamLocation>
+  {
+    AttackParamLocation.Body, AttackParamLocation.Query, AttackParamLocation.Fragment
+  };
+
+  private static readonly IEnumerable<TestType> TestTypeWhiteList = Enum
+    .GetValues(typeof(TestType))
+    .Cast<TestType>();
+
+  private static readonly IEnumerable<AttackParamLocation> AttackParamLocationWhiteList = Enum
+    .GetValues(typeof(AttackParamLocation))
+    .Cast<AttackParamLocation>();
+
+  private Target.Target _target = null!;
+  private IEnumerable<TestType> _tests = null!;
+
+  private string? _name;
+  private TimeSpan? _targetTimeout;
+  private TimeSpan? _slowEpTimeout;
+  private int? _poolSize;
+  private IEnumerable<AttackParamLocation>? _attackParamLocations;
+
+  public TargetOptions Target
+  {
+    get { return _target; }
+    set
+    {
+      _target = new Target.Target(value);
+    }
+  }
+
+  public string? Name
+  {
+    get { return _name; }
```

Comment:
Use expression body where it is possible
```suggestion
    get => _name;
```